src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowMonitor.scala

```diff
@@ -56,7 +71,32 @@ class WorkflowMonitor(parent: ActorRef, pollInterval: Duration, workflow: Workfl
     }
   }
 
-  def onWorkflowDone(updateWorkflow: Workflow, txn: RawlsTransaction): Unit = {
-    // call any code to update entity model here
+  def withMethodConfig(txn: RawlsTransaction)(op: MethodConfiguration => (Workflow, Option[Map[String, Attribute]])): (Workflow, Option[Map[String, Attribute]]) = {
+    methodConfigurationDAO.get(submission.workspaceName.namespace, submission.workspaceName.name, submission.methodConfigurationNamespace, submission.methodConfigurationName, txn) match {
+      case None => (workflow.copy(status = Failed, messages = Option(workflow.messages.getOrElse(Seq.empty) :+ AttributeString(s"Could not find method config ${submission.methodConfigurationNamespace}/${submission.methodConfigurationName}, was it deleted?"))), None)
+      case Some(methodConfig) => op(methodConfig)
+    }
+  }
+
+  def onWorkflowSuccess(workflow: Workflow, completionStatus: WorkflowStatuses.WorkflowStatus, txn: RawlsTransaction): (Workflow, Option[Map[String, Attribute]]) = {
+    withMethodConfig(txn) { methodConfig =>
+      val outputs = executionServiceDAO.outputs(workflow.workflowId, authCookie).outputs
+
+      val attributes = methodConfig.outputs.map { case (outputName, attributeName) =>
+        Try {
+          attributeName.value -> outputs.getOrElse(outputName, {
+            throw new RawlsException(s"output named ${outputName} does not exist")
+          })
+        }
+      }
+
+      if (attributes.forall(_.isSuccess)) {
+        (workflow.copy(status = completionStatus), Option(attributes.map(_.get).toMap))
```

Comment:
later if we resubmit due to some error (say output was not available at time and became available again), do we need to clear messages on workflow here?